packages/server/src/cloud/gcp/storage.ts

```diff
@@ -15,18 +15,37 @@ export class GoogleCloudStorage extends BaseBinaryStorage {
     this.bucket = this.storage.bucket(bucketName);
   }
 
-  async writeFile(key: string, contentType: string | undefined, stream: BinarySource): Promise<void> {
+  /**
+   * Writes a file to Google Cloud Storage.
+   * This method now correctly handles both Readable streams and strings as input.
+   * @param key - The key (path) for the file in the bucket.
+   * @param contentType - The MIME type of the content.
+   * @param data - The content to write, either as a Readable stream or a string.
+   */
+  async writeFile(key: string, contentType: string | undefined, data: BinarySource): Promise<void> {
```

Comment:
Wait I think this interface is wrong isn't it?

Ohh are we going to have to make the `Storage` interface more generic? 🤔

Reply:
I think it's ok, yeah?

Interface definition in `packages/server/src/storage/types.ts`:

```ts
  writeFile(key: string, contentType: string | undefined, stream: BinarySource): Promise<void>;
```

I think it gets back to how `BinarySource` is different in `core` and `server`

Reply:
Yeah I was looking at the wrong definition, I thought we were importing from `@medplum/core`. LGTM